external/mdns/mdns.c

```diff
@@ -255,7 +255,7 @@ static uint8_t *uncompress_nlabel(uint8_t *pkt_buf, size_t pkt_len, size_t off)
 		return NULL;
 	}
 	// calculate length of uncompressed label
-	for (p = pkt_buf + off; *p && p < e; p++) {
+	for (p = pkt_buf + off; p + 1 < e && *p; p++) {
```

Comment:
I think, offset can never be equal or greater than pkt_len   at this stage due to if check at line no 254.  and  we may not require this check. 

lets take simple case..  pkt_buf = 2 and pkt_len = 4 and off = 3
e  = pkt_buf + pkt_len  // e = 6
p = pkt_buf + off // p = 5
so for proper working case  , p < e  ie, ( 5 < 6 ) 
But you modified to p+1 < e   ie, ( 6 < 6 ) , 

Can you double check the changes ?

Reply:
because mdns uses name compression it is possible to access outbounded.
and a dns length of RR field is 2  so it is safer to compare as above